src/lambda/wizards/samInitWizard.ts

```diff
@@ -80,6 +100,171 @@ export class DefaultCreateNewSamAppWizardContext extends WizardContext implement
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        return val ? (val.label as string) : undefined
+    }
+
+    public async promptUserForRegistry(currRegion: string, currRegistry?: string): Promise<string | undefined> {
+        const client: SchemaClient = ext.toolkitClientBuilder.createSchemaClient(currRegion)
+        const registryNames = await SchemasDataProvider.getInstance().getRegistries(currRegion, client)
+
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.schemas.registry.prompt', 'Select a Registry'),
+                value: currRegistry ? currRegistry : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: registryNames!.map(registry => ({
+                label: registry,
+                alwaysShow: registry === currRegistry,
+                description:
+                    registry === currRegistry ? localize('AWS.wizard.selectedPreviously', 'Selected Previously') : ''
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        return val ? val.label : undefined
+    }
+
+    public async promptUserForSchema(
+        currRegion: string,
+        currRegistry: string,
+        currSchema?: string
+    ): Promise<string | undefined> {
+        const client: SchemaClient = ext.toolkitClientBuilder.createSchemaClient(currRegion)
+        const schemas = await SchemasDataProvider.getInstance().getSchemas(currRegion, currRegistry, client)
+
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.schemas.schema.prompt', 'Select a Schema'),
+                value: currSchema ? currSchema : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: schemas!.map(schema => ({
```

Comment:
Will schemas  ever contain zero or one item?

If zero, the picker wouldn't be able to proceed.
If one, is it worth showing? (I think so, but wanted to bring it up)

Reply:
Unlike registryList, this one could contain zero schemas. I can check if the schemas size is zero and if yes return back to previous step. At the same time, showInformationMessage that 'No schemas found in this registry'.  Notification can be added either here or in schemasDataProvider class. Let me know what you think